File: cases/leaflet/web_api_coffee.py
'''
請先安裝:
pip install flask flask-cors requests

資料來源:
https://cafenomad.tw/
'''

from flask import Flask, request, jsonify, send_from_directory
from flask_cors import CORS
from urllib.parse import quote_plus
import requests, uuid
from pathlib import Path

# 建立 Flask 伺服器
app = Flask(__name__)

# 自動處理 CORS
CORS(app)

# Web API
URL = "https://cafenomad.tw/api/v1.2/cafes/taipei"

# 請求標頭
HEADERS = {
    "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) "
                  "AppleWebKit/537.36 (KHTML, like Gecko) "
                  "Chrome/86.0.4240.80 Safari/537.36"
}

# 自訂路由端點 (Route Endpoint)
@app.route("/cafes/taipei", methods=["GET"])
def cafes_taipei():
    # 請求資料
    r = requests.get(URL, headers=HEADERS, timeout=10)

    # 轉換回應為 JSON 格式 (將字串轉為類似 Object/Dict 的格式)
    resp = jsonify(r.json())

    # 回傳結果
    return resp


# 主程式
if __name__ == "__main__":
    # 開發測試用
    app.run(
        host="0.0.0.0", # 允許外部存取
        port=5000, # 服務 port 號
        debug=True # 除錯模式 (開發時使用，正式上線請移除或設為 False)
    )

File: cases/leaflet/web_api_coffee.php
<?php
//回應標頭 (Response Headers)
header('Access-Control-Allow-Origin: *'); //開放所有網域請求
header('Content-Type: application/json'); //告訴前端，回傳格式為 JSON (前端接到，會是物件型態)

$headers = [
    'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.80 Safari/537.36',
];

//Web API
$url = 'https://cafenomad.tw/api/v1.2/cafes/taipei';

//模擬網路請求，取得從 LINE 貼圖網站回傳的 html
$ch = curl_init(); //curl 初始化
curl_setopt($ch, CURLOPT_URL, $url); //設定 URL
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true); //若有轉址，一路轉到正常顯示的頁面
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers); //指定 Request Headers
$result = curl_exec($ch); //取得 html
curl_close($ch); //關閉 curl

//輸出 ($result 本身就是 json 字串，在檔案最上面放 response header，告訢前端 js 我回傳給你的是 object 型態)
echo $result;